src/main/java/org/elasticsearch/index/query/SimpleQueryStringBuilder.java

```diff
@@ -117,40 +247,165 @@ public SimpleQueryStringBuilder flags(SimpleQueryStringFlag... flags) {
+        this.settings.lowercaseExpandedTerms(lowercaseExpandedTerms);
         return this;
     }
 
+    /** Returns whether parsed terms should be lower cased for this query. */
+    public Boolean lowercaseExpandedTerms() {
+        return this.settings.lowercaseExpandedTerms;
+    }
+
+    /** Specifies the locale for parsing terms. Defaults to ROOT if none is set. */
     public SimpleQueryStringBuilder locale(Locale locale) {
-        this.locale = locale;
+        this.settings.locale(locale);
         return this;
     }
 
-    public SimpleQueryStringBuilder lenient(boolean lenient) {
-        this.lenient = lenient;
+    /** Returns the locale for parsing terms for this query. */
+    public Locale locale() {
+        return this.settings.locale;
+    }
+
+    /** Specifies whether query parsing should be lenient. Defaults to false. */
+    public SimpleQueryStringBuilder lenient(Boolean lenient) {
+        this.settings.lenient(lenient);
         return this;
     }
 
-    public SimpleQueryStringBuilder analyzeWildcard(boolean analyzeWildcard) {
-        this.analyzeWildcard = analyzeWildcard;
+    /** Returns whether query parsing should be lenient. */
+    public Boolean lenient() {
+        return this.settings.lenient();
+    }
+
+    /** Specifies whether wildcards should be analyzed. Defaults to false. */
+    public SimpleQueryStringBuilder analyzeWildcard(Boolean analyzeWildcard) {
+        this.settings.analyzeWildcard(analyzeWildcard);
         return this;
     }
 
+    /** Returns whether wildcards should by analyzed. */
+    public Boolean analyzeWildcard() {
+        return this.settings.analyzeWildcard();
+    }
+
+    /**
+     * Specifies the minimumShouldMatch to apply to the resulting query should
+     * that be a Boolean query.
+     */
     public SimpleQueryStringBuilder minimumShouldMatch(String minimumShouldMatch) {
         this.minimumShouldMatch = minimumShouldMatch;
         return this;
     }
 
+    /**
+     * Returns the minimumShouldMatch to apply to the resulting query should
+     * that be a Boolean query.
+     */
+    public String minimumShouldMatch() {
+        return minimumShouldMatch;
+    }
+
+    /** Returns an exception if any mandatory fields are null. */
+    @Override
+    public QueryValidationException validate() {
+        QueryValidationException validationException = null;
+
+        // Query text is required
+        if (queryText == null) {
+            validationException = QueryValidationException.addValidationError("[" + SimpleQueryStringBuilder.NAME + "] query text missing",
+                    validationException);
+        }
+
+        return validationException;
+    }
+
+    /** Returns the lucene query parsed from the parameters. */
+    @Override
+    public Query toQuery(QueryParseContext parseContext) {
+        QueryValidationException validationException = validate();
+        if (validationException != null) {
+            throw validationException;
+        }
```

Comment:
we don't need to call validate here, I know it is confusing cause we are not calling validate in the right place yet, but we will be able to do it only later on unfortunately, for now let's rely on the assumption that at some point it will be called in the right place